Makefile

```diff
@@ -0,0 +1,171 @@
+.PHONY: clean lint requirements test dist publish bump_major_version \
+bump_minor_version bump_patch_version fetch_new_major_version \
+fetch_new_minor_version fetch_new_patch_version fetch_current_version \
+bump_build_version fetch_new_build_version bump_release_version \
+fetch_new_release_version
```

Comment:
@noahweber1 I think it is a good idea to have a Makefile, however, in some of the targets I think we should go ahead and just use hatch, which, among other things, calls various clis for tasks like linting, testing, building docs, building the package, and releasing to PyPI. See https://github.com/pinellolab/DNA-Diffusion#development and https://hatch.pypa.io/dev/intro/. This is also now built into the github actions.

Reply:
I see, you are very right. Let me remove all of the endpoints that cant be covered with hatch. Maybe some custome things we still want to expose with Makefile.